### DISCUSSION TITLE
Connect Bluetooth Headset via RFID Card - Command in RFID_trigger_play

### DISCUSSION THREAD

**Discussion by cqwer05** [MiczFlor/RPi-Jukebox-RFID#1973] [Q&A] [answered]
Hello, I installed the bluetooth-sink-switch on my sons phoniebox, everything works fine. But his bluetooth Headphones don't connect to every device they are paired if it is poissible, but only to the last connected device. so if it is connected to my smartphone once, the phoniebox can't autoconnect to it. I wrote a little script, that has the bluetoothctl commands to connect it. This script works fine.

Now I have the problem, that i'm not able to create a Card-command to connect this script to a nfc-tag. I tried to change the rfid_trigger_play.sh and the rfid_trigger_play.conf, but there is just no difference in the web interface, even if I delete all entries in this file... 

my script "connect.sh"

`#!/bin/bash
bluetoothctl -- discoverable on
bluetoothctl -- agent on
bluetoothctl -- default-agent
bluetoothctl -- connect 42:06:B2:29:3F:61
bluetoothctl -- exit
`

change in the rfid_trigger_play.sh

`        $CMDBLUETOOTHCONNECT)
            #Verbindet Bennis Kopfhoerer
            $PATHDATA/connect.sh
            ;;
`
rfid_trigger_play_conf

`##Testversuch  ### Verbindet Bennis Kopfhoerer
CMDBLUETOOTHCONNECT="14022315011206"`

i also changed the rfid_trigger_play.conf.sample, so the entry shows up in the webinterface and i can connect a card, but the function isn't executed...

is there anything in the Wiki or any thread, that i didn't find yet? I tried this thing for many hours now and looked everywhere but didn't really find a solution, and it is very disappointing right now, not getting such an easy task to work :(

**Comment by cqwer05** [comment, marked as answer]
in the next step, i tried to use the playout_controls.sh to create a function # bluetoothconnect, while the connect.sh script is in the same folder

` bluetoothconnect)
        sh connect.sh
        ;;
`
rfid_trigger_play.sh

`$CMDBLUETOOTHCONNECT)
            #Verbindet Bennis Kopfhoerer
            $PATHDATA/playout_controls.sh -c=bluetootconnect
            ;;`

I really have no idea, why my script isn't started


the output from "debug.log":

```
Card  ID '26159164144177' was used
# Attempting to play: /home/pi/RPi-Jukebox-RFID/shared/audiofolders/
# Type of play $VALUE:
Path not found /home/pi/RPi-Jukebox-RFID/shared/audiofolders/
```

i don't understand, why the phoniebox is looking for an audiofolder in this case.

**Reply by AlvinSchiller** [reply]
Hi,
this log messages are always shown if a command is used and don't indicate an error, though they are confusing cause the code block after is skipped.
Regarding you problem i have some questions to your code:

- You wrote "This script works fine". How did you call it? With "./connect.sh" or "sh connect.sh".
Because in "rfid_trigger_play.sh" you called the script with "$PATHDATA/connect.sh". 
-> does your script has execution rights (chmod +x connect.sh)?
In the next example for "playout_controls.sh" you used "sh connect.sh"
-> there is a typo in the command parameter for "playout_controls.sh" ("bluetootconnect" instead of "bluetoot**h**connect")

- Have you tried running the trigger from the terminal (`/home/pi/RPi-Jukebox-RFID/scripts/rfid_trigger_play.sh -i=14022315011206`) and see if there a error messages in the output?

- Have you added some log output to your script to see if it is really not running?
For example `echo "test start connect.sh"` if run from terminal
or `. ${PROJROOTPATH}/settings/debugLogging.conf` and `if [ "${DEBUG_rfid_trigger_play_sh}" == "TRUE" ]; then echo "test start connect.sh" >> $PATHDATA/../logs/debug.log; fi` to see output in the logfile on RFID use 
(DEBUG_rfid_trigger_play_sh must be enabled in settings).

**Reply by cqwer05** [reply]
thank you, it works now. the execution rights and the typo were the mistake.

**Reply by AlvinSchiller** [reply]
Good to hear that it works 👍

And just for reference there is a quite similiar addition: https://github.com/MiczFlor/RPi-Jukebox-RFID/discussions/1960#discussion-4751879

**Reply by s-martin** [reply]
You can mark the reply as the correct answer.
